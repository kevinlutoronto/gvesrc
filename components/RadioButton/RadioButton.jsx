// @flow
import React from 'react';
import cx from 'classnames';
import Icon from '../Icon/Icon';
import { getDataAttrs } from '../../dataUtils';
import type { Data } from '../../types';
import style from './style.scss';

type Props = {|
  id?: string,
  data?: Data,
  checked?: boolean,
  description?: ?string,
  disabled?: ?boolean,
  fancy?: ?boolean,
  fieldValue?: string,
  icon?: ?string,
  isLarge?: ?boolean,
  name: string,
  onChange: (event: SyntheticEvent<HTMLInputElement>) => mixed,
  text: React$Node,
  value?: string,
|};

class RadioButton extends React.Component<Props> {
  static defaultProps = {
    name: 'radioButton',
    text: 'Option',
  };

  value = (): mixed => {
    const { checked, value, fieldValue } = this.props;
    return checked || (fieldValue && value === this.props.fieldValue);
  };

  render() {
    const {
      data,
      description,
      disabled,
      fancy,
      fieldValue,
      icon,
      isLarge,
      name,
      onChange,
      text,
      id,
    } = this.props;
    const wrapperClasses = cx(style.wrapper, {
      [style.disabled]: disabled,
      [style.fancy]: fancy,
      [style.fancyChecked]: fancy && this.value(),
      [style.isLarge]: isLarge,
    });
    const inlineClass = cx({ [style.inline]: fancy });
    const fauxCircle = fancy ? null : <div className={style.faux} />;
    const fancyIcon =
      fancy && icon ? (
        <div className={style.fancyIcon}>
          <Icon icon={icon} />
        </div>
      ) : null;
    return (
      <div {...getDataAttrs(data)} className={inlineClass}>
        <label htmlFor={id} className={wrapperClasses}>
          <input
            className={cx(style.original, {
              [style.disabled]: disabled,
            })}
            checked={this.value()}
            disabled={disabled}
            id={id}
            name={name}
            onChange={onChange}
            type="radio"
            value={fieldValue}
          />
          {fauxCircle}
          {fancyIcon}
          <div
            className={cx(style.text, {
              [style.disabled]: disabled,
            })}
          >
            {text}
          </div>
        </label>
        {description && <span className={style.description}>{description}</span>}
      </div>
    );
  }
}

RadioButton.displayName = 'Plasma@RadioButton';

export default RadioButton;
