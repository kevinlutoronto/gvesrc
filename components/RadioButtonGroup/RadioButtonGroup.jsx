// @flow
import React from 'react';
import RadioButton from '../RadioButton/RadioButton';

type Option = {|
  description?: string,
  disabled?: boolean,
  icon?: string,
  label: string,
  value: string,
|};

type Props = {|
  fancy?: boolean,
  isLarge?: boolean,
  name: string,
  onChange: () => mixed,
  options: Array<Option>,
  value?: string,
|};

class RadioButtonGroup extends React.Component<Props> {
  renderRadioButtons = (): React$Node => {
    const { options, value, fancy, isLarge, name, onChange } = this.props;
    // TODO: replace fieldValue with value={opt.value}
    return (
      options &&
      options.map((opt: Option) => (
        <RadioButton
          checked={value === opt.value}
          description={opt.description}
          disabled={opt.disabled}
          icon={opt.icon}
          key={opt.value}
          text={opt.label}
          fieldValue={opt.value}
          fancy={fancy}
          isLarge={isLarge}
          name={name}
          onChange={onChange}
        />
      ))
    );
  };

  render() {
    return <div>{this.renderRadioButtons()}</div>;
  }
}

RadioButtonGroup.displayName = 'Plasma@RadioButtonGroup';

export default RadioButtonGroup;
