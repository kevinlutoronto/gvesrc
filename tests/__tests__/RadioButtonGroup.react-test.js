/* eslint-disable */
import React from 'react';
import RadioButton from '../../components/RadioButton/RadioButton';
import RadioButtonGroup from '../../components/RadioButtonGroup/RadioButtonGroup';
import renderer from 'react-test-renderer';

describe('RadioButtonGroup Component', () => {
  test('render', () => {
    const tree = renderer
      .create(<RadioButtonGroup name="test" options={[]} onChange={() => {}} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
