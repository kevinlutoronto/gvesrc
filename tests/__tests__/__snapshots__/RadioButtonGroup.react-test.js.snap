// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButtonGroup Component render 1`] = `<div />`;
